ddtrace/propagation/http.py

```diff
@@ -106,6 +126,34 @@ def _inject(span_context, headers):
         if span_context.dd_origin is not None:
             headers[HTTP_HEADER_ORIGIN] = ensure_text(span_context.dd_origin)
 
+        if not config._x_datadog_tags_enabled:
+            span_context._meta["_dd.propagation_error"] = "disabled"
+
+        # Do not try to encode tags if we have already tried and received an error
+        if "_dd.propagation_error" in span_context._meta:
+            return
+
+        # Only propagate tags that are allowed for propagation
+        tags_to_encode = {
+            k: ensure_str(v)
+            for k, v in ((ensure_str(k), v) for k, v in span_context._meta.items())
+            if k in _DatadogMultiHeader._X_DATADOG_TAGS_ALLOW
+        }  # type: Dict[str, str]
```

Comment:
I put a comment in the code for why we need to coerce.

If we want to avoid this, we would need to change the typing of the meta dictionary in `Context` vs `Span`.